### PR TITLE
Additional Known Parameters for Aspire Bicep files

### PR DESCRIPTION
**Pull request by rudiv**
This is a mini change to allow the use of 2 additional known parameters from Aspire's Bicep tool which adds some more flexibility to custom Bicep:

- environmentName
- tags

When using infra synth the output expected is:

```
module customresource 'my-custom-bicep/custom.bicep' = {
  name: 'customresource'
  scope: rg
  params: {
    location: location
    environmentName: environmentName
    principalId: resources.outputs.MANAGED_IDENTITY_PRINCIPAL_ID
    tags: tags
  }
}
```

This allows us to use the environment name in automated deployments between test/live given the lack of support within azd currently. See https://github.com/Azure/azure-dev/discussions/3184 for more info on that discussion.

Note: I haven't added this to Aspire's KnownParameters as some new CDK is supposedly changing how that works, but I'm assuming that it will stay relatively similar on the azd side.

### CODE REVIEW

**Comment by rudiv** [comment]
@microsoft-github-policy-service agree [company="divergent Cloud Ltd"]

**Comment by rudiv** [comment]
@microsoft-github-policy-service agree company="divergent Cloud Ltd"

**Review by vhvb1989 — APPROVED** [review]
I'm fine with the change, but I need ack from @mitchdenny and @davidfowl . The known vars should be in sync with Aspire.

**Comment by rudiv** [comment]
I can add them to the Aspire side if needed but will hold off until confirmed useful / needed, it is from my side so just running this custom build for now and passing as the strings within the AppHost.

**Review by rajeshkamal5050 — CHANGES_REQUESTED** [review]
Do not merge - https://github.com/Azure/azure-dev/pull/3627#issuecomment-2030209612

**Comment by vhvb1989** [comment]
environmentName has been added here: https://github.com/Azure/azure-dev/pull/3933

The only missing one for this request would be the `tags`

**Comment by rudiv** [comment]
I’m a little confused how both were added here, and half of it has been done elsewhere? @vhvb1989 

Tags were to bring it in to parity with the other list, they’re not particularly necessary for the use case (which was environment name). Shall I just close this off now?

**Comment by vhvb1989** [comment]
> I’m a little confused how both were added here, and half of it has been done elsewhere? @vhvb1989
> 
> Tags were to bring it in to parity with the other list, they’re not particularly necessary for the use case (which was environment name). Shall I just close this off now?

My bad.  
In #3933, we have added two new known parameters, but it is the Azure Containers Environment (ACE) id and name.
And reading this issue again, the ask was not for the ACE name, but for the dotner environment name?

**Comment by rudiv** [comment]
Nope, it was for the environment passed in to AZD as part of setting up an Aspire project - so technically this still isn't added.

It's captured as a parameter in main.bicep:

```
@minLength(1)
@maxLength(64)
@description('Name of the environment that can be used as part of naming resource convention, the name of the resource group for your application will use this name, prefixed with rg-')
param environmentName string
```

.. and then subsequently used for the Resource Group creation:

```
resource rg 'Microsoft.Resources/resourceGroups@2022-09-01' = {
  name: 'rg-${environmentName}'
  location: location
  tags: tags
}
```

My change allowed you to pass this down to other added resources so that you could detect based on that name whether you're in a certain environment. For example, if you called your environment "dev" (or something containing it), provision free/low cost resources, "stage" low cost, "live" production ready. It doesn't really matter as the choice was always yours in how it was handled.

This worked around the azd limitation of not being able to have different sets of infra for different environments, as you could just put it all in the same bicep file and switch based on the environment that you were in. It was also a super quick win (as we're literally just saying it's there, pass it on deployment via azd).

**Pull request closed without merging**
